tests/Middleware/RedirectTest.php

```diff
@@ -0,0 +1,129 @@
+<?php
+
+namespace Yiisoft\Yii\Web\Tests\Middleware;
+
+use Nyholm\Psr7\Factory\Psr17Factory;
+use Nyholm\Psr7\Response;
+use Nyholm\Psr7\ServerRequest;
+use PHPUnit\Framework\TestCase;
+use Psr\Http\Message\ResponseInterface;
+use Psr\Http\Message\ServerRequestInterface;
+use Psr\Http\Server\RequestHandlerInterface;
+use Yiisoft\Router\Method;
+use Yiisoft\Router\UrlGeneratorInterface;
+use Yiisoft\Yii\Web\Middleware\Redirect;
+
+final class RedirectTest extends TestCase
+{
+    /**
+     * @test
+     */
+    public function invalidArguments()
+    {
+        $this->expectException(\InvalidArgumentException::class);
+        $this->createRedirectMiddleware()->process($this->createRequest(), $this->createRequestHandler());
+    }
+
+    /**
+     * @test
+     */
+    public function generateUri()
+    {
+        $middleware = $this->createRedirectMiddleware()
+            ->toRoute('test/route', [
+                'param1' => 1,
+                'param2' => 2,
+            ]);
+
+        $response = $middleware->process($this->createRequest(), $this->createRequestHandler());
+        $header = $response->getHeader('Location');
+
+        $this->assertSame($header[0], 'test/route?param1=1&param2=2');
+    }
+
+    /**
+     * @test
+     */
+    public function temporaryReturnCode302()
+    {
+        $middleware = $this->createRedirectMiddleware()
+            ->toRoute('test/route')
+            ->temporary();
+
+        $response = $middleware->process($this->createRequest(), $this->createRequestHandler());
+
+        $this->assertSame($response->getStatusCode(), Redirect::TEMPORARY);
```

Comment:
I'd not rely on Redirect constants in tests. That would prevent modifying constant values.